proxy/mempool/mempool_schedule.py

```diff
@@ -150,6 +152,7 @@ def len_tx_nonce_queue(self) -> int:
     def add_tx(self, tx: MPTxRequest) -> None:
         assert self._state_tx_cnt <= tx.nonce, f'Tx {tx.sig} has nonce {tx.nonce} less than {self._state_tx_cnt}'
         self._tx_nonce_queue.add(tx)
+        self._heartbeat = int(time())
```

Comment:
The sorted queue doesn't automatically update the sort.
So, the correct actions should happen on the add tx to mempool (MPTxSchedule::_add_tx_to_sender_pool):
- remove the pool from the heartbeat queue
- update heartbeat
- add the pool to the heartbeat queue

Reply:
I am now removing and adding the sender pool within `MPTxSchedule::_add_tx_to_sender_pool()` and letting `add_tx()` update the heartbeat.

However, I have had to gate the `pop()` with a `find()` because I think some tests are removing the sender pool too early.